packages/node/src/sdk/init.ts

```diff
@@ -90,13 +92,20 @@ export function init(options: NodeOptions | undefined = {}): void {
   }
 
   if (!isCjs()) {
-    // We want to make sure users see this warning
-    consoleSandbox(() => {
-      // eslint-disable-next-line no-console
-      console.warn(
-        '[Sentry] You are using the Sentry SDK with an ESM build. This version of the SDK is not compatible with ESM. Please either build your application with CommonJS, or use v7 of the SDK.',
-      );
-    });
+    const [nodeMajor, nodeMinor] = process.versions.node.split('.').map(Number);
+
+    // Register hook was added in v20.6.0 and v18.19.0
+    if (nodeMajor > 22 || (nodeMajor === 20 && nodeMinor >= 6) || (nodeMajor === 18 && nodeMinor >= 19)) {
+      // @ts-expect-error register is available in these versions & import.meta is allowed
+      moduleModule.register('@opentelemetry/instrumentation/hook.mjs', import.meta.url);
```

Comment:
just to be sure, does it matter if this is registered twice?

Reply:
Ok, I tested this and registering twice is a problem. We get around this for ourselves by only registering once and tracking with a global whether we have already registered.

I can see this becoming a problem if people have their own hooks registered already. I currently don't see a straight forward way of detecting this so for now we will just have to document this behaviour.

We should keep an eye open though for automatic detection.